#!/bin/bash
#SBATCH --time=23:46:00
#SBATCH --mem=4GB
#SBATCH -p long
# Set number of nodes to run
#SBATCH --nodes=1
# Set number of tasks to run
#SBATCH --ntasks=1
# Set number of cores per task (default is 1)
#SBATCH --cpus-per-task=8
# Output and error files
#SBATCH -o job.%J.out

source activate Genomics

rm -r /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/DepthFinder
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/Genomic
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/ultVCFs
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/AlleleFreq
mkdir /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/AlleleFreq/Summary


samtools depth -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/BAMs/Sort/Bismark-Bowtie.sbS100X.deduplicated-sorted.bam | awk 'BEGIN{OFS=FS="\t"; bases=0; print "Chromosome" OFS "Basepairs" OFS "Depth"}NR==1{chrom=$1; depth=$3; bases=1; next}$1!=chrom{print chrom OFS bases OFS depth/bases; chrom=$1; depth=$3; bases=1; next}{depth=depth+$3; bases++}END{print chrom OFS bases OFS depth/bases}' >  /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/DepthFinder/gtSub_ShermanSim100X.chrom-depth.txt &
samtools depth -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/BAMs/Sort/Bismark-Bowtie.sbS100X.deduplicated-sorted.bam | awk 'BEGIN{OFS=FS="\t"; depth=0}{depth=depth+$3}END{print depth/NR}' > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/DepthFinder/gtSub_ShermanSim100X.depth.txt &
wait


(awk 'BEGIN{OFS=FS="\t"}NR<3{next}{print $0}' /work/mjuswilc/Domain/GreatTits/General/Annotations/Abel/Repeats/RepeatMasker/RepeatMasker-DFAM3.7/Abel-GCF_001522545.3_Parus_major1.1_genomic.fna.out; awk 'BEGIN{OFS=FS="\t"}NR<3{next}{print $0}' /work/mjuswilc/Domain/GreatTits/General/Annotations/Abel/Repeats/RepeatMasker/RepeatMasker-RM2.0.4DFAM3.7/GCF_001522545.3_Parus_major1.1_genomic.fna.out) | awk 'BEGIN{OFS="\t"}NR<=3{next}{print $5 OFS $6 OFS $7}' | sort -k1,1V -k2,2n | uniq | awk 'BEGIN{OFS=FS="\t"}NR==1{scaff=$1; start=$2; end=$3; next}$1==scaff && ($2<end && $3<end){next}$1==scaff && ($2<end && $3>end){end=$3;next}$1!=scaff || $2>end{print scaff OFS start OFS end; scaff=$1; start=$2; end=$3}END{print scaff OFS start OFS end}' > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/Genomic/GCF_001522545.3_Parus_major1.1.repeats.bed &
awk 'BEGIN{OFS="\t"}{print $1 OFS $2 OFS $3; print $4 OFS $5 OFS $6}' /work/mjuswilc/Domain/GreatTits/General/Annotations/Abel/SegmentalDuplications/BISER/Output/Abel-GCF_001522545.3_Parus_major1.1_genomic.fna.seg_dup.bedpe > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/Genomic/GCF_001522545.3_Parus_major1.1.seg_dup.bed &
hDEPTH=`cat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/DepthFinder/gtSub_ShermanSim100X.depth.txt | awk '{print int(($0*2)+0.5)}'`
lDEPTH=`cat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/DepthFinder/gtSub_ShermanSim100X.depth.txt | awk '{print int(($0*0.5)+0.5)}'`
zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | grep -v "#" | perl -pe 's#(\S+)\s+(\d+).*ADF=(\d+),(\d+);ADR=(\d+),(\d+).*#$1\t$2\t$3\t$4\t$5\t$6#' | awk 'BEGIN{OFS=FS="\t"}2*($3+$4)<($5+$6){next}2*($5+$6)<($3+$4){next}{print $1 OFS $2 OFS $2}' > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.read-symetrical.bed &
zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk 'BEGIN{OFS=FS="\t"}/^#/{next}$6<30 || $7=="Low"{print $1 OFS $2 OFS $2}' > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.low-quality.bed &
zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | grep -v "#" | perl -pe 's#(\S+)\s+(\d+).*ADF=(\d+),(\d+);ADR=(\d+),(\d+).*#$1\t$2\t$3\t$4\t$5\t$6#' | awk -v low=$lDEPTH -v high=$hDEPTH 'BEGIN{OFS=FS="\t"}{depth=$3+$4+$5+$6}depth<low{print $1 OFS $2 OFS $2; next}depth>high{print $1 OFS $2 OFS $2; next}' > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.low-depth.bed &
awk 'BEGIN{OFS=FS="\t"}NR==1{next}{print $1 OFS $2 OFS $2+1}' /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X.CG > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.CG.bed &
zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk 'BEGIN{OFS=FS="\t"}/^#/{next}length($5)>1{print $1 OFS $2 OFS $2}' > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.multi-allelic.bed &
wait


function VCFerate {
(zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk '!/^#/{exit}{print}'; bedtools subtract -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/Genomic/GCF_001522545.3_Parus_major1.1.repeats.bed) | bgzip -l 9 > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.no-repeats.vcf.gz &
(zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk '!/^#/{exit}{print}'; bedtools subtract -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/Genomic/GCF_001522545.3_Parus_major1.1.seg_dup.bed) | bgzip -l 9 > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.no-seg_dups.vcf.gz &
(zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk '!/^#/{exit}{print}'; bedtools intersect -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.read-symetrical.bed) | bgzip -l 9 > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.read-symetrical.vcf.gz &
(zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk '!/^#/{exit}{print}'; bedtools subtract -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.low-quality.bed) | bgzip -l 9 > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.Q30-pass.vcf.gz &
(zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk '!/^#/{exit}{print}'; bedtools subtract -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.low-depth.bed) | bgzip -l 9 > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.high-depth28X.vcf.gz &
(zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk '!/^#/{exit}{print}'; bedtools subtract -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.CG.bed) | bgzip -l 9 > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.nCpG.vcf.gz &
(zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz | awk '!/^#/{exit}{print}'; bedtools subtract -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/SNPs/sbS100X-SNV.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/BEDS/ShermanSim100X.snp.vcf.multi-allelic.bed) | bgzip -l 9 > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.no-multi-allelic.vcf.gz &
wait
for v in $(ls /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.*.vcf.gz)
do
bcftools index $v &
done
wait
bcftools isec -o /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.fullFilt.vcf -n 7 -w 1 /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.no-repeats.vcf.gz /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.no-seg_dups.vcf.gz /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.read-symetrical.vcf.gz /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.Q30-pass.vcf.gz /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.high-depth28X.vcf.gz /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.nCpG.vcf.gz /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.no-multi-allelic.vcf.gz
(awk '!/^#/{exit}{print}' /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.fullFilt.vcf; awk 'BEGIN{OFS=FS="\t"}/^#/{print $0; next}$4=="A" && $5=="T"{print $0; next}$4=="T" && $5=="A"{print $0; next}$4=="C" && $5=="A"{print $0; next}$4=="G" && $5=="T"{print $0; next}' /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/filtVCFs/ShermanSim100X.snp.fullFilt.vcf) > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/ultVCFs/ShermanSim100X.snp.fullFilt.vcf
bgzip -l 9 /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/ultVCFs/ShermanSim100X.snp.fullFilt.vcf
bcftools index /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/ultVCFs/ShermanSim100X.snp.fullFilt.vcf.gz
}

VCFerate
echo 'Filtration Files Created!!!'

echo -e "Chromosome\tPosition\tRefAllele\tAltAllele\tMutation\tPhred\tDepth\tAlleleFreq" > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/AlleleFreq/Summary/sbS100X-SNV-AlleleFreq.txt
zcat /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/ultVCFs/ShermanSim100X.snp.fullFilt.vcf.gz | grep -v "#" | perl -pe 's#(\S+)\s+(\S+)\s+\.\s+(\S+)\s+(\S+)\s+(\S+).*DP=(\d+).*(\d\.\d+)$#$1\t$2\t$3\t$4\t$3$4\t$5\t$6\t$7#' >> /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/AlleleFreq/Summary/sbS100X-SNV-AlleleFreq.txt

echo -e "Chromosome\tPosition\tRefAllele\tAltAllele\tMutation\tPhred\tDepth\tAlleleFreq" > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/AlleleFreq/Summary/sbS100X-SNV-TrueHetAlleleFreq.txt
bedtools intersect -f 1 -F 1 -wa -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/ultVCFs/ShermanSim100X.snp.fullFilt.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/SimulatedGenome/Haplotype2/great-tit-sim.Abel-GCF_001522545v3-Hap2_ms.vcf | grep -v "#" | perl -pe 's#(\S+)\s+(\S+)\s+\.\s+(\S+)\s+(\S+)\s+(\S+).*DP=(\d+).*(\d\.\d+)$#$1\t$2\t$3\t$4\t$3$4\t$5\t$6\t$7#' >> /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/AlleleFreq/Summary/sbS100X-SNV-TrueHetAlleleFreq.txt

echo -e "Chromosome\tPosition\tRefAllele\tAltAllele\tMutation\tPhred\tDepth\tAlleleFreq" > /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/AlleleFreq/Summary/sbS100X-SNV-ErrorAlleleFreq.txt
bedtools subtract -A -a /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/ultFilter/ultVCFs/ShermanSim100X.snp.fullFilt.vcf.gz -b /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/SimulatedGenome/Haplotype2/great-tit-sim.Abel-GCF_001522545v3-Hap2_ms.vcf | grep -v "#" | perl -pe 's#(\S+)\s+(\S+)\s+\.\s+(\S+)\s+(\S+)\s+(\S+).*DP=(\d+).*(\d\.\d+)$#$1\t$2\t$3\t$4\t$3$4\t$5\t$6\t$7#' >> /work/mjuswilc/Domain/GreatTits/ultDeconvolution/HeterozygoteSimulation/GreatTit/Sherman/AlleleFilt/AlleleFreq/Summary/sbS100X-SNV-ErrorAlleleFreq.txt

echo 'Filtered Allele Frequecny Spectra Extracted!!!'
